### DISCUSSION TITLE
Logs not visible in SkyWalking UI after configuring Kafka-fetcher

### DISCUSSION THREAD

**Discussion by abhishekacharya828** [apache/skywalking#12112] [General]
I am using Filebeat to send logs to a Kafka topic. After that, I want to visualize those logs in the SkyWalking server. I have successfully sent logs to the Kafka topic using Filebeat, and I have also configured the Kafka-fetcher in SkyWalking to receive logs from the Kafka topic. However, the logs are not visible in the SkyWalking server UI.

Could you please help me troubleshoot this issue? I would be happy to provide any additional information or logs that might be helpful in resolving this problem.

Thank you in advance for your assistance.

**Comment by wu-sheng** [comment]
https://github.com/apache/skywalking/blob/0b9bbb47b3bedb78d016f9654c2f447c69b3bf7f/oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/provider/handler/LogHandler.java#L71-L87

According to codes, Kafka fetcher only supports proto objects, I am not sure how you do that with filebeats. I remember only HTTP receiver supports JSON format, which is supported by Filebeat

**Reply by abhishekacharya828** [reply]
If you need to review any configurations on my local system, please let me know. Additionally, I require assistance with configuring log fetching in SkyWalking. I have followed the official documentation but encountered issues. Could you please advise on the basic configuration for this ?

**Reply by wu-sheng** [reply]
I explained it, do you follow? I mean it is not supported through your way.

**Comment by abhishekacharya828** [comment]
Yes, I understand. It seems that my approach is not supported. Could you please provide guidance on the appropriate method to achieve this ?

**Reply by wu-sheng** [reply]
You could use filebeat HTTP to send JSON.

https://skywalking.apache.org/docs/main/latest/en/api/log-data-protocol/#http-api

**Comment by abhishekacharya828** [comment]
As per your instructions, logs are sent directly to the SkyWalking port, bypassing Kafka. However, I would like the logs to go through a pipeline like Filebeat => Kafka => SkyWalking. It's worth noting that this is an HTTP API, and Filebeat does not support sending data in this manner.

Filebeat.yml
![Screenshot from 2024-04-16 16-03-32](https://github.com/apache/skywalking/assets/134144944/d41106dc-db0d-438f-8b0d-b27f4f365bd2)

output section
![Screenshot from 2024-04-16 16-04-02](https://github.com/apache/skywalking/assets/134144944/026c2cf3-aa7e-4fac-a0dd-f89559b828ef)

**Reply by wu-sheng** [reply]
Filebeat is not my field. I can't help on how to set that up

SkyWalking provided a HTTP/1 endpoint with JSON format. That is all.

**Reply by wu-sheng** [reply]
Fluentbit has offical doc, https://docs.fluentbit.io/manual/pipeline/outputs/skywalking, but filebeat doesn't.

**Reply by abhishekacharya828** [reply]
I want to send logs using Filebeat to Kafka and visualize the logs in SkyWalking, as mentioned in your official documentation. I have followed the documentation but encountered problems. Please provide me with the proper procedure to achieve this. 
In official document it mention also.

![Screenshot from 2024-04-16 17-05-30](https://github.com/apache/skywalking/assets/134144944/66243e29-548c-4e10-a5d3-efe991e1f873)

**Reply by wu-sheng** [reply]
I said, it is not supported.

**Reply by abhishekacharya828** [reply]
If filebeat  is not supported then why you mentioned it in official documentaiton  ?

**Reply by wu-sheng** [reply]
I said, HTTP endpoint works. Kafka works for logs, just not filebeat. That is filebeat limitation.

**Reply by abhishekacharya828** [reply]
Can  I use fluentd to transfer logs to skywalking ?

**Reply by wu-sheng** [reply]
No matter what agent you use, most of them are JSON based, and require HTTP endpoint.

If you insist Kafka, we have to extend codes of OAP to add Kafka+JSON raw text support. I am afraid, it is not supported yet.

**Reply by abhishekacharya828** [reply]
can i directly send logs to skywalking server by  using    logstash  ?

**Reply by wu-sheng** [reply]
Wasn't I saying clearly about APIs? 
We don't care what you are using to collect, you only pick HTTP or Kafka(Protobuf data).
